### PR TITLE
Add endgame skill and 0.18.0 test plans (duzitong tasks)

### PR DESCRIPTION
**Pull request by duzitong**
## Motivation

The endgame verification workflow was living as a standalone `.github/prompts/endgame.prompt.md` file that couldn't be invoked as a first-class agent skill. This PR converts it into a proper skill and also delivers the 0.18.0 test plans for the tasks assigned to @duzitong.

### Endgame skill (`endgame`)

Converts `endgame.prompt.md` into `.github/skills/endgame/SKILL.md` with:

- YAML frontmatter (`name`, `description`) matching the `ui-action` skill structure so it is discoverable and invocable
- Simplified 4-step workflow: ask for inputs -> fetch issue (one `gh issue view` call) -> delegate each task to a subagent -> summarize
- Subagent template writes test plans to `test-plans/<slug>/<slug>.md`, following the project format exemplified by `test-plans/thinking-persistence/`
- `manage_todo_list` dependency removed (no longer needed)

### SWTBot smoke-test probe

`endgame-chat-open-001.json` validates that the Copilot Chat view opens, the model picker resolves, and the chat input is ready. All 14 steps pass locally.

### 0.18.0 test plans (4 files)

| File | Covers |
|---|---|
| `test-plans/skills-prompt-files/` | TC-001-009: Enable Skills preference, workspace/global skill discovery, prompt files as slash commands, Ask-mode exclusion, auto-refresh, prefix filtering, persistence across restarts |
| `test-plans/cls-session-persistence/` | TC-002-009: Conversation history display, Eclipse restart persistence, new-conversation isolation, transcript directory, cancelled-turn restoration, 400 Bad Request regression (tool-call message ordering + duplicate content) |
| `test-plans/subagent-event-leak/` | TC-001-003, TC-005: Subagent progress events isolated by `conversationId` across session switches and new chats |
| `test-plans/subagent-restoration/` | TC-001-004: Subagent turns nested inside parent turn after restore, including cancelled turns |

Test cases that were not observable from the UI (internal implementation details, LLM quality judgements, timing-dependent races) were excluded to keep the plans actionable.

### CODE REVIEW

File: test-plans/subagent-event-leak/subagent-event-leak.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+#### Preconditions
+- Copilot chat is open in Agent mode
+- At least one prior conversation exists in chat history (or a second conversation can be started)
+
+#### Steps
+1. Send a message in Agent mode that triggers a subagent execution (e.g. "create a new file and write a hello world program")
+2. Observe the subagent progress spinner appearing in the chat UI
+3. While the subagent is still running (spinner visible), open the chat history panel
+4. Select a different (previously existing) conversation to switch to it
+5. Observe the chat UI for the newly opened conversation while the subagent continues running in the background
```
Comment (jdneo) [inline test-plans/subagent-event-leak/subagent-event-leak.md]: The premise here doesn't match the current behavior. Switching to a history conversation (or clicking New Chat) calls ChatView.clearCurrentConversation() → onCancel(), which sends conversation/destroy for subagentConversationId and cancels all conversationFutures ([ChatView.java#L1249-L1296](https://github.com/microsoft/copilot-for-eclipse/blob/4bc02ddef34e1dd04971542f5b5d57bd5757bd09/com.microsoft.copilot.eclipse.ui/src/com/microsoft/copilot/eclipse/ui/chat/ChatView.java#L1249-L1296)). The subagent does not keep running in the background.

So:

TC-001 Step 5 / Expected Result — "subagent continues running… completes without corrupting" is unreachable.
TC-003 — switching back will show the cancelled subagent turn, not completed results.
TC-005 — New Chat also cancels the subagent via the same path.
This is already covered correctly by the existing com.microsoft.copilot.eclipse.swtbot.test/test-plans/subagent/subagent.md (TC-003 / TC-004), which expects conversation/destroy on switch.

Suggest either rewriting these cases to verify cancellation-on-switch, or dropping this file in favor of the existing subagent.md.

File: com.microsoft.copilot.eclipse.swtbot.test/test-plans/subagent-restoration/subagent-restoration.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
+# Subagent Turns Appear as Separate Assistant Messages After Restoration
```
Comment (jdneo) [inline com.microsoft.copilot.eclipse.swtbot.test/test-plans/subagent-restoration/subagent-restoration.md]: Please also check this md contains any duplication with subagent.md, and let's merge them into the existing subagent.md

----
From AI

`test-plans/subagent-restoration/subagent-restoration.md`

overlaps with the existing file:

`com.microsoft.copilot.eclipse.swtbot.test/test-plans/subagent/subagent.md`

### Duplicated / overlapping coverage

The new endgame file covers these scenarios:

- completed subagent restores nested inside parent turn
- multiple subagents restore nested correctly
- no-subagent conversation is unaffected
- cancelled / partially streamed subagent restores nested correctly

The existing SWTBot test plan already covers most of that:

| New endgame file | Existing SWTBot plan |
|---|---|
| `TC-001: Subagent turn restores nested inside parent agent turn` | `TC-002: Subagent content restored after session switch (completed)` |
| `TC-002: Multiple subagent turns all restore nested correctly` | `TC-005: Two subagents in a single turn restore correctly` |
| `TC-004: Partially-streamed subagent turn (cancelled) restores correctly` | `TC-003: Subagent content restored after cancel mid-execution` |

### What is unique in the new file

Only one case appears clearly unique:

- `TC-003: Conversation with no subagent is unaffected`

That regression case is not directly present in the existing SWTBot subagent plan.

File: com.microsoft.copilot.eclipse.swtbot.test/test-plans/cls-session-persistence/cls-session-persistence.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
+# CLS Session Persistence and Restoration for Conversation History
+
+## Overview
+This feature integrates CLS (Copilot Language Server) server-side session persistence so that when a conversation is restored the CLS receives the original `conversationId` and `restoreToTurnId`, enabling full context continuation rather than relying solely on IDE-side turn history. It also configures `~/.copilot/eclipse` as the transcript directory (matching IntelliJ's convention).
+
```
[the remark this answers is not in the document]
Comment (duzitong) [reply com.microsoft.copilot.eclipse.swtbot.test/test-plans/cls-session-persistence/cls-session-persistence.md:7]: Fixed -- changed all occurrences of \~/.copilot/eclipse\ to \<user home>/.copilot/eclipse\, with a note in the Overview explaining the equivalent Windows path \%USERPROFILE%\.copilot\eclipse\.

File: test-plans/subagent-event-leak/subagent-event-leak.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+#### 📸 Key Screenshots
+- [ ] **After return** — Original conversation with complete subagent output intact
+
+---
+
+### TC-005: New conversation started while subagent runs in background
+
+**Type:** `Edge Case`
+**Priority:** `P1`
+
```
[the remark this answers is not in the document]
Comment (duzitong) [reply test-plans/subagent-event-leak/subagent-event-leak.md]: Dropping this file entirely based on @jdneo's feedback below -- the premise was wrong (switching sessions cancels the subagent via conversation/destroy, not lets it run in the background), and these scenarios are already covered correctly in the existing `com.microsoft.copilot.eclipse.swtbot.test/test-plans/subagent/subagent.md` (TC-003, TC-004).

File: .github/skills/endgame/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+---
+
+### Step 2 — Fetch the endgame issue (ONE call only)
+
+```shell
+gh issue view <issue_number> --repo microsoft/copilot-for-eclipse
+```
+
+Parse the body to find **all tasks (checkboxes) assigned to the specified
+user**. Extract each task's title and any linked PR/issue URL as plain text.
```
[the remark this answers is not in the document]
Comment (duzitong) [reply .github/skills/endgame/SKILL.md:41]: Updated -- the `gh issue view` call now uses `--json body --jq '.body'` for reliable, parseable output.

File: .github/skills/endgame/SKILL.md [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+description: Orchestrate endgame verification for a GitHub milestone issue. Fetches the issue, parses assigned tasks, delegates each task to a subagent that researches the linked PR/issue and writes a test plan, and saves every plan to test-plans/ following the project's standard test-plan format.
+---
+
+# Endgame Verification Skill
+
+Use this skill to run an endgame verification pass for a GitHub milestone issue.
+Each task in the issue is delegated to a subagent that researches the linked
+PR/issue and writes a test plan under `test-plans/`.
+
+## Workflow
```
[the remark this answers is not in the document]
Comment (duzitong) [reply .github/skills/endgame/SKILL.md:12]: Deleted `.github/prompts/endgame.prompt.md` -- the skill is now the single source of truth.

File: test-plans/subagent-event-leak/subagent-event-leak.md (hunk 1 above)
Comment (duzitong) [reply test-plans/subagent-event-leak/subagent-event-leak.md]: Agreed -- dropped `test-plans/subagent-event-leak/subagent-event-leak.md` entirely. The premise was based on a misunderstanding; since `clearCurrentConversation() -> onCancel()` cancels the subagent on switch, there is no 'leak' scenario to verify. The scenarios that matter (cancellation on switch, no leaked output) are already correctly covered by the existing `subagent.md` TC-003 and TC-004.

File: com.microsoft.copilot.eclipse.swtbot.test/test-plans/subagent-restoration/subagent-restoration.md (hunk 2 above)
Comment (duzitong) [reply com.microsoft.copilot.eclipse.swtbot.test/test-plans/subagent-restoration/subagent-restoration.md]: Done -- TC-001/002/004 duplicated existing subagent.md TC-002/005/003. Merged the one unique case (TC-003: conversation with no subagent restores cleanly) as TC-006 into subagent.md, and deleted subagent-restoration.md.

**Review by jdneo — APPROVED** [review]
